app/mailers/notifier.rb

```diff
@@ -94,11 +94,14 @@ def diary_comment_notification(comment, recipient)
       @readurl = diary_entry_url(comment.diary_entry.user, comment.diary_entry, :anchor => "comment#{comment.id}")
       @commenturl = diary_entry_url(comment.diary_entry.user, comment.diary_entry, :anchor => "newcomment")
       @replyurl = new_message_url(comment.user, :message => { :title => "Re: #{comment.diary_entry.title}" })
-
+      @ref = "osm-diary-#{comment.diary_entry.id}@#{Settings.server_url}"
       @author = @from_user
 
       attach_user_avatar(comment.user)
 
+      headers["In-Reply-To"] = @ref
+      headers["References"]  = @ref
```

Comment:
I think we could reduce the amount of duplication here by having a `set_references` helper function and then just doing `set_references("diary", comment.diary_entry)` etc in each of the three methods?

Reply:
Good point, added a new helper method now.